public/app/plugins/datasource/loki/modifyQuery.ts

```diff
@@ -167,7 +174,9 @@ export function addLabelToQuery(query: string, key: string, operator: string, va
   const filter = toLabelFilter(key, value, operator);
   // If we have non-empty stream selector and parser/label filter, we want to add a new label filter after the last one.
   // If some of the stream selectors don't have matchers, we want to add new matcher to the all stream selectors.
-  if (everyStreamSelectorHasMatcher && (labelFilterPositions.length || parserPositions.length)) {
+  if (forceAsLabelFilter) {
+    return addFilterAsLabelFilter(query, streamSelectorPositions, filter);
```

Comment:
Also, I suspect that there is a reason to add this right after selector and not use `positionToAdd`, so would you add a comment that explains it.